src/mjolnir/pbfgraphparser.cc

```diff
@@ -616,15 +616,20 @@ struct graph_callback : public OSMPBF::Callback {
       } */
       else if (tag.first == "official_name" && !tag.second.empty()) {
         w.set_official_name_index(osmdata_.name_offset_map.index(tag.second));
-
       } else if (tag.first == "max_speed") {
         try {
-          max_speed = std::stof(tag.second);
+          if (tag.second == "unlimited") {
+            // this way has an unlimited speed limit (german autobahn)
+            max_speed = kUnlimitedSpeedLimit;
+          } else {
+            max_speed = std::stof(tag.second);
+          }
```

Comment:
This is a change from master. 

Previously edges tagged with `maxspeed=none` would not have had max_speed or tagged_speed set because `none` is non-numeric, so the road speed would fallback to the classified speed for the road class. Now that we give it a numeric value (255) the road have a default speed of 140 (max). 

I _think_ this is an improvement given that the speeds on these unlimited speed limit highways tend to be higher than the average highway. But I can also rearrange this to leave the default speed unchanged. What do you think @gknisely ?

Reply:
I would return the max classified `105` if it is `255`  @dnesbitt61 @dgearhart  thoughts?

Reply:
I would allow a higher speed for unlimited...150'ish seems better to me (somewhere near 90MPH). I'm sure there are vehicles moving faster than that, but wonder what the average is?

Reply:
should we have `one speed` for `unlimited` or should we have `one speed` per road class

Reply:
From [wikipedia](https://en.wikipedia.org/wiki/Speed_limits_in_Germany):
```
In sections of the motorway system without speed limit it is common to be overtaken by cars or motorcycles travelling over 200 km/h (125 mph). However, a recommended speed limit (Richtgeschwindigkeit) of 130 km/h (81 mph) applies. 
```
So maybe `165` kph?

Reply:
Since there's some uncertainty about how best to assign default speed in the case of maxspeed=none, I opted to back out that change and leave the default speed how it was before. That is, roads with maxspeed=none will be considered untagged, and use classified speed. https://github.com/valhalla/valhalla/pull/2251/commits/9d6c333bd321719a3e2838fef02e5ab2cbdcb7c8